### ISSUE TITLE
Support jQuery Mobile alpha 3

### ISSUE THREAD

**Issue by GoogleCodeExporter** [erhanselman/mobile-web-development-with-phonegap#13]
```
Alpha 3 released Feb 3.

```

Original issue reported on code.google.com by `paul.beu...@gmail.com` on 4 Feb 2011 at 9:22

**Comment by GoogleCodeExporter** [comment]
Original comment by `paul.beu...@gmail.com` on 4 Feb 2011 at 9:23
- Added labels: **Milestone-Release1.2.0**

**Comment by GoogleCodeExporter** [comment]
```
There is a parser issue on a questionable JavaScript regular expression in 
jquery.mobile-1.0a3.js

A workaround is to change line 1711 from
            if (href.search(/^[^:/]+:\/\/[^/]+\/?/) == -1){
to
            if (href.search(/^[^:\/]+:\/\/[^\/]+\/?/) == -1){

Opened https://github.com/jquery/jquery-mobile/issues/issue/1010 about the 
JavaScript and https://bugs.eclipse.org/bugs/show_bug.cgi?id=336579 about 
JSDT's poor error handling

The minimized version also doesn't parse. More investigation needed. A 
workaround is to rename the file to jquery.mobile-1.0a3.min.js1

```

Original comment by `paul.beu...@gmail.com` on 8 Feb 2011 at 4:30
- Changed title: **Support jQuery Mobile alpha 3**
- Added labels: **Type-Defect**
- Removed labels: **Type-Enhancement**

**Comment by GoogleCodeExporter** [comment]
```
I found that pointing the wizard at my own alpha3 (part of the regular jQuery 
Mobile project creation) installation resulted in an incorrect index.html file. 
Review it before using and the fixes should be obvious (e.g. 1.0a2 referenced 
instead of 1.0a3 for some files).

You must also import jquery.1.5 (should be used with the mobile alpha 3) and 
change index.html accordingly, as the wizard currently installs jquery.1.4.4.
```

Original comment by `libby.ba...@gmail.com` on 12 Feb 2011 at 9:27

**Comment by GoogleCodeExporter** [comment]
```
Multiple a3 and phonegap issues are being discussed in the jquery mobile 
forums.  See http://forum.jquery.com/topic/jquerymobile-1-0a3-errors-in-code 
and 
http://forum.jquery.com/topic/switching-from-alpha2-to-alpha3-with-android-local
-navigation-problem
```

Original comment by `paul.beu...@gmail.com` on 13 Feb 2011 at 4:35

**Comment by GoogleCodeExporter** [comment]
```
Because of the issues with a3, MDS 1.2 will stick with a2 as the default 
jquery.mobile installation.

To use a3 with MDS 1.2, download a3, and choose set its location in the project 
wizard.

Then, update in whichever of jquery.mobile-1.0a3.js (line 1711) and/or 
jquery.mobile-1.0a3.min.js(line 43) the string /^[^:\/]+:\/\/[^\/]+\/?/ to 
/^[^:/]+:\/\/[^/]+\/?/

Also, stick with jquery-1.4.4.js instead of 1.5 if you need file: web links 
(like in the included jqm UI demo)
```

Original comment by `paul.beu...@gmail.com` on 23 Feb 2011 at 1:24
- Removed labels: **Milestone-Release1.2.0**

**Comment by GoogleCodeExporter** [comment]
```
Now that alpha 4 is out, we won't do an alpha 3 integration. This bug will be 
closed once 1.2.2 is released with a4 support. 
```

Original comment by `paul.beu...@gmail.com` on 5 Apr 2011 at 4:00

**Comment by GoogleCodeExporter** [comment]
```
a4.1 is bundled with MDS 1.2.2
```

Original comment by `paul.beu...@gmail.com` on 12 Apr 2011 at 4:22
- Changed state: **WontFix**
